drivers/display/ek79652.c

```diff
@@ -0,0 +1,725 @@
+/*
+ * Copyright (c) 2021 Linumiz
+ *
+ * Based on gd7965.c which is:
+ * Copyright (c) 2020 PHYTEC Messtechnik GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT gooddisplay_ek79652
+
+#include <string.h>
+#include <drivers/display.h>
+#include <drivers/gpio.h>
+#include <drivers/spi.h>
+#include <sys/byteorder.h>
+
+#include "ek79652_regs.h"
+
+#define LOG_LEVEL CONFIG_DISPLAY_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(ek79652);
+
+/**
+ * EK79652 compatible EPD controller driver.
+ *
+ */
+
+#define EK79652_PIXELS_PER_BYTE		8U
+
+struct ek79652_dt_array {
+	uint8_t *data;
+	uint8_t len;
+};
+
+struct ek79652_config {
+	struct spi_dt_spec bus;
+	struct gpio_dt_spec reset;
+	struct gpio_dt_spec dc;
+	struct gpio_dt_spec busy;
+	struct ek79652_dt_array softstart;
+	struct ek79652_dt_array pwr;
+	struct ek79652_dt_array pwropt;
+	struct ek79652_dt_array lut_vcom_dc;
+	struct ek79652_dt_array lut_ww;
+	struct ek79652_dt_array lut_wb;
+	struct ek79652_dt_array lut_bw;
+	struct ek79652_dt_array lut_bb;
+	uint16_t height;
+	uint16_t width;
+	uint8_t cdi;
+};
+
+struct ek79652_data {
+	bool initial;
+	bool full_refresh;
+	bool disp_refresh;
+};
+
+static int ek79652_write_cmd(const struct device *dev,
+				   uint8_t cmd, uint8_t *data, size_t len)
+{
+	const struct ek79652_config *config = dev->config;
+	struct spi_buf buf = {.buf = &cmd, .len = sizeof(cmd)};
+	struct spi_buf_set buf_set = {.buffers = &buf, .count = 1};
+	int err;
+
+	err = gpio_pin_set_dt(&config->dc, 1);
+	if (err < 0) {
+		return err;
+	}
+
+	err = spi_write_dt(&config->bus, &buf_set);
+	if (err < 0) {
+		return err;
+	}
+
+	if (data != NULL) {
+		for (int i = 0 ; i < len ; i++) {
```

Comment:
This loop does not make any sense for me, why cannot you reuse `struct spi_buf buf` like gd7965 driver or define a new one `struct spi_buf foo = {.buf = data, .len = len};` ?

>Because according to [datasheet](https://ibb.co/FDBjLKQ) inbetween sending bytes CS pin should be made HIGH. Since CS pin need to be made LOW only when sending data, this display is working only when sending byte by byte data. In zephyr we can't toggle CS pin. so only ek79652_write_data() function is introduced.

I doubt that.